meteor/client/lib/guiSettings/studioSettings.tsx

```diff
@@ -0,0 +1,274 @@
+import { StudioBaselineStatus } from '../../ui/Settings/Studio/Baseline'
+import { useHistory } from 'react-router-dom'
+import { MeteorCall } from '../../../lib/api/methods'
+
+export function generateStudioSettings(t: TFunction, studio: Studio): GUISettings {
+	const settings: GUISettings = { list: [] }
+
+	settings.list.push({
+		type: GUISettingsType.SECTION,
+		name: 'Generic Properties',
+		id: 'generic',
+
+		getList: () => genericProperties(t, studio, 'generic'),
+		getSearchString: '',
+		// renderSummary?: () => JSX.Element
+	})
+
+	// settings.list.push({
+	// 	type: GUISettingsType.SECTION,
+	// 	name: 'Attached Devices',
+
+	// 	getList: () => (GUISetting | GUISettingSection)[]
+	// 	// renderSummary?: () => JSX.Element
+	// })
+
+	// settings.list.push({
+	// 	type: GUISettingsType.SECTION,
+	// 	name: 'Blueprint Configuration',
+
+	// 	getList: () => (GUISetting | GUISettingSection)[]
+	// 	// renderSummary?: () => JSX.Element
+	// })
+
+	// settings.list.push({
+	// 	type: GUISettingsType.SECTION,
+	// 	name: 'Layer mappings',
+
+	// 	getList: () => (GUISetting | GUISettingSection)[]
+	// 	// renderSummary?: () => JSX.Element
+	// })
+	// settings.list.push({
+	// 	type: GUISettingsType.SECTION,
+	// 	name: 'Route sets',
+
+	// 	getList: () => (GUISetting | GUISettingSection)[]
+	// 	// renderSummary?: () => JSX.Element
+	// })
+
+	// settings.list.push({
+	// 	type: GUISettingsType.SECTION,
+	// 	name: 'Package Manager',
+
+	// 	getList: () => (GUISetting | GUISettingSection)[]
+	// 	// renderSummary?: () => JSX.Element
+	// })
+
+	return settings
+}
+
+function getDefaultEditAttributeProps(studio: Studio) {
+	return literal<Partial<IEditAttribute>>({
+		...defaultEditAttributeProps,
+		obj: studio,
+		collection: Studios,
+	})
+}
+
+function genericProperties(t: TFunction, studio: Studio, urlBase: string) {
+	const settings: GUISetting[] = []
+
+	const editAttributeProps = getDefaultEditAttributeProps(studio)
+
+	settings.push({
+		type: GUISettingsType.SETTING,
+		name: t('Studio Name'),
+		description: t('Name of the studio'),
+		id: `${urlBase}/name`,
+		getWarning: () => {
+			return !studio.name && t('No name set')
+		}, // json {requires: []}, minLength: 1, maxLength: 255
+		render: () => <EditAttribute {...editAttributeProps} attribute="name" type="text" />,
+		getSearchString: studio.name,
+		// ui-widget: textEdit
+	})
+
+	// Note: Non-reactive when called directly:
+	const getAvailableShowStyleBases = () => {
+		return ShowStyleBases.find(
+			{},
+			{
+				projection: {
+					_id: 1,
+					name: 1,
+				},
+			}
+		)
+			.fetch()
+			.map((showStyle) => {
+				return {
+					_id: showStyle._id,
+					name: `${showStyle.name}`,
+					value: showStyle._id,
+				}
+			})
+	}
+	settings.push({
+		type: GUISettingsType.SETTING,
+		name: t('Compatible Show Styles'),
+		description: t('Which Show Styles are compatible with this studio'),
+		id: `${urlBase}/blueprint`,
```

Comment:
It would be really good if these ID's were autogenerated somehow? I think these will be very easy to get mixed up.